### ISSUE TITLE
add substitutions

### ISSUE THREAD

**Issue by FedericoTartarini** [jupyter-book/mystmd#1551]
It would be great if we could add substitutions to the front matter so we can reference them in the paper.

## Add substitutions

Adding the ability to include substitutions in the document formatted so we can then reference these keys in the md file. 

### Example

```
myst:
  substitutions:
    key1: "I'm a **substitution**"
```

Then in the document we can reference the keys as.

This is the text {{ key1 }} I want to substitute.

Similar to what is implemented in the [myst-parser](https://myst-parser.readthedocs.io/en/latest/syntax/optional.html#substitutions-with-jinja2_

**Comment by choldgraf** [comment]
Agreed this would be helpful! We've got an issue tracking this at the link below so I'll close this one as a duplicate but please +1 the idea below!

- https://github.com/jupyter-book/mystmd/issues/852
